java/arcs/core/data/ParticleSpec.kt

```diff
@@ -22,5 +22,13 @@ data class ParticleSpec(
     val claims: List<Claim> = emptyList(),
     val checks: List<Check> = emptyList(),
     /** Indicates whether the particle is an isolated (non-egress) particle. */
-    val isolated: Boolean = false
-)
+    val isolated: Boolean = false,
+    /** Optional egress type of the particle. Must be null for isolated particles. */
+    val egressType: String? = null
```

Comment:
maybe it's better to add an annotations list to ParticleSpec class, and have getters for `isolated` and `egressType`?

Reply:
Done